src/gui/uithememanager.cpp

```diff
@@ -133,7 +133,14 @@ QIcon UIThemeManager::getIcon(const QString &iconId, const QString &fallback) co
 QIcon UIThemeManager::getFlagIcon(const QString &countryIsoCode) const
 {
     if (countryIsoCode.isEmpty()) return {};
-    return QIcon(":/icons/flags/" + countryIsoCode.toLower() + ".svg");
+
+    const auto iter = m_flagCache.find(countryIsoCode);
+    if (iter != m_flagCache.end())
+        return *iter;
+
+    const QIcon icon {QLatin1String(":/icons/flags/") + countryIsoCode.toLower() + QLatin1String(".svg")};
+    m_flagCache[countryIsoCode] = icon;
```

Comment:
`countryIsoCode` should be normalized (such as lowercase) before you use it as a key to store value.